src/libraries/System.Security.Cryptography.Xml/src/System/Security/Cryptography/Xml/EncryptedXml.cs

```diff
@@ -204,7 +204,7 @@ private byte[] GetCipherValue(CipherData cipherData)
                 if (cipherData.CipherReference.Uri.Length == 0)
                 {
                     // self referenced Uri
-                    string baseUri = _document?.BaseURI;
+                    string baseUri = _document?.BaseURI!;
```

Comment:
This is a strange construction... if _document is null, baseUri will be null.  Why isn't it `string? baseUri`?  Maybe there are `?`s missing elsewhere?

Reply:
I remember it seeming strange when typing it actually! `baseUri` can't be nullable, because it's the last parameter in `TransformChain::TransformToOctetStream`, which _isn't_ nullable.

But, digging through a bit more, this value ulimately ends up being stored in `Transform::BaseURI`, which **is** nullable.

I'll change the parameters to be nullable.